### PR TITLE
added publish method to document manager

### PR DESCRIPTION
**Pull request by danrot**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Fixed tickets | fixes #2028 |
| Related issues/PRs | #issuenum |
| License | MIT |
| Documentation PR | TODO |
#### What's in this PR?

This PR adds a publish method to the `DocumentManager`.
#### Why?

Because we have to publish documents for the drafting feature.

### CODE REVIEW

File: lib/DocumentManagerInterface.php [hunk 1]
```diff
@@ -60,7 +60,7 @@ public function remove($document);
      * Move the PHPCR node to which the document is mapped to be a child of the node at the given path or UUID.
      *
      * @param object $document
-     * @param string $destId Path or UUID
```
Comment (danrot) [inline lib/DocumentManagerInterface.php:63]: Maybe we should change that to only path... The problem is that not every node has a UUID (e.g. or maybe only the newly created ones), but every node has a path (also the ones which are only existing in the current session).

Comment (chirimoya) [inline lib/DocumentManagerInterface.php:63]: ???

File: lib/Event/PublishEvent.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    public function __construct($document, $locale)
+    {
+        $this->document = $document;
+        $this->locale = $locale;
+    }
+
+    /**
+     * Sets the node this event should operate on.
+     *
+     * TODO Check if should be move to DocumentManager itself
```
Comment (chirimoya) [inline lib/Event/PublishEvent.php]: issue

**Comment by chirimoya** [comment]
ACK

File: lib/Event/PublishEvent.php (hunk 2 above)
Comment (danrot) [inline lib/Event/PublishEvent.php]: https://github.com/sulu/sulu-document-manager/issues/94

File: lib/DocumentManagerInterface.php (hunk 1 above)
Comment (danrot) [inline lib/DocumentManagerInterface.php:63]: Fixed comment

**Pull request merged**